app_check/src/ios/debug_provider_ios.mm

```diff
@@ -26,11 +67,19 @@
   return g_debug_app_check_provider_factory;
 }
 
-DebugAppCheckProviderFactory::DebugAppCheckProviderFactory() {}
+DebugAppCheckProviderFactory::DebugAppCheckProviderFactory() {
+  if (!g_ios_debug_app_check_provider_factory) {
+    g_ios_debug_app_check_provider_factory = [[FIRAppCheckDebugProviderFactory alloc] init];
+  }
+}
 
 DebugAppCheckProviderFactory::~DebugAppCheckProviderFactory() {}
 
-AppCheckProvider* DebugAppCheckProviderFactory::CreateProvider(App* app) { return nullptr; }
+AppCheckProvider* DebugAppCheckProviderFactory::CreateProvider(App* app) {
+  FIRAppCheckDebugProvider* createdProvider =
+      [g_ios_debug_app_check_provider_factory createProviderWithApp:app->GetPlatformApp()];
+  return new internal::DebugAppCheckProvider(createdProvider);
```

Comment:
I realize this is potentially leaking memory.  The cleanup for these things is sorta confusing.  We should probably use a map to save the returned providers, which will also let us return the same Provider for the same App (which based on the documentation it should be doing.)

Reply:
I updated device check and app attest to also have an internal class and I updated all three ios providers to have a map from app* to provider*